code_challenges/code_wars/find_the_parity_outlier.py

```diff
@@ -0,0 +1,20 @@
+# https://www.codewars.com/kata/5526fc09a1bbd946250002dc
+
+
+def parity_outlier(numbers):
+    even_int = 0
+    odd_int = 0
+
+    for num in numbers:
+        if num % 2 == 0:
+            even_int += 1
+            if even_int > 1:
+                for num in numbers:
+                    if num % 2 != 0:
+                        return num
+        else:
+            odd_int += 1
+            if odd_int > 1:
+                for num in numbers:
+                    if num % 2 == 0:
+                        return num
```

Comment:
This is good. Always remember that the number one goal when doing these on the spot is to get to a working solution.  Then refactor to something better if time warrants.

No need to change what you have though, just food for thought below.

Since the outside for loop is doing one thing and the inside a different it might be better to split them.  Really ones personal preference on this one I think.
```python
def parity_outlier(numbers):
    even_int = 0
    odd_int = 0
    for num in numbers[:3]:
        if num % 2 == 0:
            even_int += 1
        else:
            odd_int += 1
             
    if even_int > 1:
        for num in numbers:
            if num % 2 != 0:
                return num
    else:
        for num in numbers:
            if num % 2 == 0:
                return num
```

Since the code in the if/else blocks are basically the same, here is a way to reuse the same code.
```python
def parity_outlier(numbers):
    are_even = [num % 2 == 0 for num in numbers[:3]].count(True) > 1
    outlier_modulus = 0 if are_even else 1
    for num in numbers:
        if num % 2 != outlier_modulus:
            return num
```

The `are_even = ` line above isn't great.  Not the most readable.  Your method is a-okay, I just wanted to show a usage you might not have seen yet. Creating a list from expressions that evaluate into booleans.  It doesn't substantially increase the codes big O complexity so if one can stomach the looks it's probably okay.

Reply:
let me try to put that into words

`num % 2 == 0 for num in numbers[:3]`
return a true or false for the first three nums in numbers

then the count(True) iterates over are_even to find the count of True?

But what exactly is happening at the end with the > 1??

I mean, my brain can make logic of what it is DOING but I don't grasp how that actually works. I must be looking at it wrong.

I see a list comprehension: 
`are_even = [num % 2 == 0 for num in numbers[:3]]`

but I have not experienced the rest .count(True) > 1 

I guess what I am really asking is how do you just throw a > 1 on the end and it still work??

Also, this would only check for the first three nums in numbers so if the parity outlier was the 86th number it wouldn't find it right? 

Either way, I am very excited for that one line!!! The more I can get out of comprehensions the better!!!

Reply:
> But what exactly is happening at the end with the > 1??

Lets break down the whole thing.
```python
numbers = [160, 3, 1719, 19, 11, 13, -21]
are_even =   [num % 2 == 0 for num in numbers[:3]].count(True) > 1
                                      numbers[:3]
                                    [160, 3, 1719]
           [num % 2 == 0 for num in [160, 3, 1719]]
           [True, False, True]
           [True, False, True]                    .count(True)
           2
           2                                                   > 1
are_even = True
```

[list.count(x)](https://docs.python.org/3/tutorial/datastructures.html)

> I guess what I am really asking is how do you just throw a > 1 on the end and it still work??

Everything evaluates into it's simplest form.

> Also, this would only check for the first three nums in numbers so if the parity outlier was the 86th number it wouldn't find it right?

How many numbers does one need to check to know if the list if _of_ even or odd numbers?